logoapp/js/dot_data.js

```diff
@@ -0,0 +1,1028 @@
+	setInterval(this.loop.bind(this) , 33 );
+	this.drawButton();
+};
+DotAppFramework.prototype.copyArrDotFrame = function(arrDotFrame) {
+	var newArray = new Array(8);
+		for(var i=0; i<newArray.length; i++) {
+			newArray[i] = arrDotFrame[i].copy();
+	}
+	return newArray;
+};
+DotAppFramework.prototype.unshiftHistoryArrDotFrame = function() {
+	this.historyArrDotFrame.unshift(this.copyArrDotFrame(this.arrDotFrame));
+};
+DotAppFramework.prototype.onClickMain = function(e) {
+	var rect = e.target.getBoundingClientRect();
+	var x = e.clientX - rect.left;
+	var y = e.clientY - rect.top;
+	this.mainView.mousePos = new Vector2(x, y);
+};
+DotAppFramework.prototype.onMouseDownMain = function() {
+	this.isPressMain = true;
+
+	if(this.isDraw) {
+		if(this.isHoverDot) {
+			if(this.hoverDotNum==this.preDrawNum) {
+				if(this.arrDotFrame[this.currentFrame-1].arrDot[this.hoverDotNum].isDraw) {
+					this.arrDotFrame[this.currentFrame-1].arrDot[this.hoverDotNum].isDraw = false;
+				}
+				else {
+					this.arrDotFrame[this.currentFrame-1].arrDot[this.hoverDotNum].isDraw = true;
+					this.arrDotFrame[this.currentFrame-1].arrDot[this.hoverDotNum].color = this.currentColor;
+				}
+			}
+			this.addHistory();
+			this.buttonActive.clear=false;
+			for(var i=0; i<DOT_NUM; i++) {
+				if(this.arrDotFrame[this.currentFrame-1].arrDot[i].isDraw) {
+					this.buttonActive.clear=true;
+					break;
+				}
+			}
+		}
+		this.isDrawHover=true;
+	}
+}
+DotAppFramework.prototype.onMouseUpMain = function() {
+	this.isPressMain = false;
+	this.isDrawHover = false;
+}
+DotAppFramework.prototype.addHistory = function() {
+	if(this.historyPos!= 0) {
+		for(var i=0; i<this.historyPos; i++) {
+			this.historyArrDotFrame.shift();
+		}
+		this.historyPos=0;
+	}
+	this.unshiftHistoryArrDotFrame();
+	if(this.historyArrDotFrame.length>20) {
+		this.historyArrDotFrame.pop();
+	}
+	this.buttonActive.undo=true;
+	this.drawButton();
+};
+DotAppFramework.prototype.onClickPreview = function(e) {
+	var clickNum;
+	for(var i=0; i<8; i++) {
+		if(this.previewView[i].isMouseOnCanvas) {
+			clickNum=i;
+		}
+	}
+	if(this.isCopy) {
+		if(clickNum!=this.currentFrame-1) {
+			this.isCopy=false;
+			this.arrDotFrame[clickNum] = this.arrDotFrame[this.currentFrame-1].copy();
+			this.currentFrame = clickNum+1;
+			this.arrIsAccess[this.currentFrame-1]=true;
+			this.addHistory();
+			for(var i=0; i<DOT_NUM; i++) {
+				if(this.arrDotFrame[this.currentFrame-1].arrDot[i].isDraw) {
+					this.buttonActive.clear=true;
+					break;
+				}
+			}
+			var isComplete = true;
+			for(var i=0; i<this.arrIsAccess.length; i++) {
+				if(this.arrIsAccess[i] == false) {
+					isComplete = false;
+					break;
+				}
+			}
+			this.buttonActive.complete = isComplete;
+
+			this.buttonActive.copy = true;
+			this.buttonActive.play = true;
+		}
+	}
+	else {
+		this.currentFrame = clickNum+1;
+		this.arrIsAccess[this.currentFrame-1]=true;
+		var isComplete = true;
+		for(var i=0; i<this.arrIsAccess.length; i++) {

```

Comment:
__[jshint]__
*  'i' is already defined.


<!-- Comment by [SideCI](https://sideci.com) -->